pkg/llmstrategy/react/strategy.go

```diff
@@ -251,37 +252,37 @@ type Message struct {
 	Content string `json:"content"`
 }
 
-// Data represents the structure of the data to be filled into the template.
-type Data struct {
+// PromptData represents the structure of the data to be filled into the template.
+type PromptData struct {
 	Query       string
 	PastQueries string
-	History     string
+	History     []Message
 	Tools       string
 }
 
 // generateFromTemplate generates a prompt for LLM. It uses the prompt from the provides template file or default.
-func (a *Strategy) generatePrompt(_ context.Context, defaultPromptTemplate string, data Data) (string, error) {
+func (a *Strategy) generatePrompt(_ context.Context, defaultPromptTemplate string, data PromptData) (string, error) {
 	var tmpl *template.Template
 	var err error
+	var contentStr string
 
 	if a.PromptTemplateFile != "" {
 		// Read custom template file
 		content, err := os.ReadFile(a.PromptTemplateFile)
 		if err != nil {
 			return "", fmt.Errorf("error reading template file: %v", err)
 		}
-		tmpl, err = template.New("customTemplate").Parse(string(content))
-		if err != nil {
-			return "", fmt.Errorf("error parsing custom template: %v", err)
-		}
+		contentStr = string(content)
 	} else {
 		// Use default template
-		tmpl, err = template.New("promptTemplate").Parse(defaultPromptTemplate)
-		if err != nil {
-			return "", err
-		}
+		contentStr = defaultPromptTemplate
+	}
+	contentStr = strings.ReplaceAll(contentStr, "JSON_BLOCK_START", "```json")
```

Comment:
Do we do this because of the problem escaping backticks in go strings?  Because if so, maybe let's just embed the prompt (using go:embed)